jina/serve/runtimes/gateway/http_fastapi_app_docarrayv2.py

```diff
@@ -187,8 +187,10 @@ def add_post_route(
         )
         app_kwargs['response_class'] = DocArrayResponse
 
+        from fastapi import Request
+
         @app.api_route(**app_kwargs)
-        async def post(body: input_model, response: Response):
+        async def post(body: input_model, response: Response, request: Request):
```

Comment:
Does this means the header will only be available when gateway is enabled?

Reply:
Anyway - having access to HTTP header is very important for our production application - we are an infra team supporting different businesses, and we use headers to communicate extra information like auth, tracing id, etc.

@JoanFM

Reply:
It should work for standalone-fast-api deployments too see [this change](https://github.com/jina-ai/jina/pull/6126/files#diff-76f86adabfbe802643bffc1fcd78a47e594e330aa0c0d3e787c166d21abb46caR92).

Reply:
hey @cenhao ,

The feature should be available with and without gateway.

Btw, would be cool to know about ur usecase and how u use/plan to use Jina